iris-mpc-gpu/tests/e2e.rs

```diff
@@ -1015,28 +998,118 @@ mod e2e_test {
                     }
                     TestCases::ReauthNonMatchingTarget => {
-                        tracing::info!("Sending reauth request non-matching the target index");
-                        let (db_index, _) = self.get_iris_code_in_db(DatabaseRange::Full);
+                        tracing::info!(
+                            "Sending reauth request with AND rule non-matching the target index"
+                        );
+                        let (db_index, _) = self.get_iris_code_in_db(DatabaseRange::FullMaskOnly);
                         self.db_indices_used_in_current_batch.insert(db_index);
                         self.reauth_target_indices
                             .insert(request_id.to_string(), db_index as u32);
-                        let template = IrisCode::random_rng(&mut self.rng);
+
+                        // prepare a template that matches only on one side
+                        // it will end up with a failed reauth with AND rule
+                        self.or_rule_matches.push(request_id.to_string());
+                        let will_match = true;
+                        let flip_right = Some(self.rng.gen());
+                        let template = self.prepare_flipped_codes(db_index, will_match, flip_right);
                         self.expected_results
                             .insert(request_id.to_string(), ExpectedResult {
                                 db_index:             None,
                                 is_batch_match:       false,
                                 is_reauth_successful: Some(false),
                             });
-                        E2ETemplate {
-                            left:  template.clone(),
-                            right: template,
-                        }
+                        template
+                    }
+                    TestCases::ReauthOrRuleMatchingTarget => {
+                        tracing::info!(
+                            "Sending reauth request with OR rule matching the target index"
+                        );
+                        let (db_index, _) = self.get_iris_code_in_db(DatabaseRange::FullMaskOnly);
+                        self.db_indices_used_in_current_batch.insert(db_index);
+                        self.disallowed_queries.push(db_index as u32);
+                        self.or_rule_matches.push(request_id.to_string());
+                        self.reauth_target_indices
+                            .insert(request_id.to_string(), db_index as u32);
+                        or_rule_indices = vec![db_index as u32];
+                        let will_match = true;
+                        let flip_right = Some(self.rng.gen());
+                        let template = self.prepare_flipped_codes(db_index, will_match, flip_right);
+                        self.expected_results
+                            .insert(request_id.to_string(), ExpectedResult {
+                                db_index:             None,
```

Comment:
If we expect this to match, shouldn't the `db_index` be set to the `db_index` from which we get the template?